src/utils/warnings.js

```diff
@@ -1,43 +1,73 @@
-// utils
+import { warn as print } from './logger'
 
 const log = function() {
-  var ref
-  try {
-    return (ref = window.console) != null
-      ? typeof ref.log === 'function'
-        ? ref.log(...arguments)
+  if (process.env.NODE_ENV !== 'production') {
+    print(
+      "Please don't use `log` function.\n" +
+        'Support for it will be removed in the next major release.'
```

Comment:
I think it would be better in this way:
`It's support will be removed in the next major release.` or just
`It will be removed in the next major release.`

Reply:
I see the same text in react-router, so origin one is also ok I think.